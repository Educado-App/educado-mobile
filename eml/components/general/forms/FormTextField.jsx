import React from 'react';
import { View, TextInput } from 'react-native';
import Text from '../Text';
import PropTypes from 'prop-types';

/**
 * Text field component for forms (e.g. login, register, etc.). 
 * @param {Obj} props Possible props:
 * - label: Label to be displayed
 * - required: Whether the field is required
 * - placeholder: Placeholder text to be displayed
 * - keyboardType: Keyboard type (e.g. numeric, email-address, etc.)
 * - autoComplete: Whether to enable auto-completion
 * - secureTextEntry: Whether to mask the input (for passwords, etc.)
 * - passwordGuidelines: Whether to display password guidelines
 * - onChangeText: Callback function to be called when the text changes
 * - value: Value of the input
 * - bordered: Whether to display a border
 * - error: Whether to display an error border
 * @returns {React.Element} Text component for entering data
 */
export default function FormTextField(props) {
	return (
		<View>
			<View className="flex flex-row">
				<Text className={'ml-2 text-xs'}>
					{props.label ?? ''}
				</Text>
				<Text className={'ml-1 text-xs text-error'}>
					{props.required ? '*' : ''}
				</Text>
			</View>
			<View className=''>
				<TextInput className={'h-50 br-25 py-1 pl-[10px] bg-projectWhite rounded-lg' +
          (props.bordered ? ' border-2 border-projectGray' : '') + (props.error ? ' border-2 border-error' : '')}
				placeholder={props.placeholder ?? ''} // Placeholder text to be displayed
				keyboardType={props.keyboardType ?? 'default'} // Keyboard type (e.g. numeric, email-address, etc.)
				autoComplete={props.autoComplete ?? 'off'} // Whether to enable auto-completion
				secureTextEntry={props.secureTextEntry ?? false} // Whether to mask the input (for passwords, etc.)
				passwordGuidelines={props.passwordGuidelines ?? false} // Whether to display password guidelines
				onChangeText={props.onChangeText ? value => props.onChangeText(value) : null} // Callback function to be called when the text changes
				value={props.value} // Value of the input
				/>
			</View>
		</View>
	);
}

FormTextField.propTypes = {
	autoComplete: PropTypes.string,
	bordered: PropTypes.bool,
	error: PropTypes.bool,
	keyboardType: PropTypes.string,
	label: PropTypes.string,
	onChangeText: PropTypes.func,
	passwordGuidelines: PropTypes.bool,
	placeholder: PropTypes.string,
	required: PropTypes.bool,
	secureTextEntry: PropTypes.bool,
	value: PropTypes.string,
};