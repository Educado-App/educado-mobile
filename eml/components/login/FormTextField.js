import React, { useState } from "react";

import { Text, View, TextInput } from "react-native";

export default function FormTextField(props) {

  const displayPasswordGuidelines = (props) => {
    if (props.passwordGuidelines) {
      return (
        <View className='flex flex-column'>
          <Text className='ml-3 text-xs text-gray mb-1'>• Mínimo 8 caracteres</Text>
          <Text className='ml-3 text-xs text-gray mb-1'>• Conter pelo menos uma letra</Text>
        </View>
      )
    }
  }


  return (
    <View className='mx-10'>
      <View className='flex flex-row'>
        <Text className='ml-2 text-xs'>{props.label ? props.label : ''}</Text>
        <Text className='ml-1 text-xs color-red-600'>{props.required ? '*' : ''}</Text>
      </View>
      <View className=''>
        <TextInput className='bg-white h-50 br-25 py-1 pl-[10px] border-gray-200 rounded-xl'
          //Phone Number
          placeholder={props.placeholder ? props.placeholder : ""}
          keyboardType={props.keyboardType ? props.keyboardType : "default"}
          autoComplete={props.autoComplete ? props.autoComplete : "off"}
          secureTextEntry={props.secureTextEntry ? props.secureTextEntry : false}
          passwordGuidelines={props.passwordGuidelines ? props.passwordGuidelines : false}
        //onChangeText={phoneNumber => setPhoneNumber(phoneNumber)}
        //value={phoneNumber}
        />
      </View>
      {displayPasswordGuidelines(props)}
    </View>

  )
}