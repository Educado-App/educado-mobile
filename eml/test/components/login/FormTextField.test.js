import renderer from "react-test-renderer";
import React from "react";
import FormTextField from "../../../components/general/forms/FormTextField";

test('Ensure that the text field is rendered correctly', async () => {
  let textField;
  await renderer.act(() => {
    textField = renderer.create(<FormTextField />);
  });
  let tree = textField.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Ensure that the field props are rendered correctly', async () => {
  let textField;
  await renderer.act(async () => {
    textField = renderer.create(<FormTextField
      placeholder="Test"
      keyboardType="default"
      secureTextEntry={false}
      onChangeText={() => { return; }}
      value="Test"
    />);
  });
  expect(textField.root.props.placeholder).toBe('Test');
  expect(textField.root.props.keyboardType).toBe('default');
  expect(textField.root.props.secureTextEntry).toBe(false);
  expect(textField.root.props.onChangeText).toBeDefined();
  expect(textField.root.props.value).toBe('Test');
});