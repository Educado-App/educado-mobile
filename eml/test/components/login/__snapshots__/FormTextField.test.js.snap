// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ensure that the text field is rendered correctly 1`] = `
<View>
  <View
    style={
      [
        {
          "display": "flex",
        },
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Montserrat-Regular",
          },
          {
            "color": "#383838",
          },
          {
            "fontSize": 16,
          },
          [
            {
              "marginLeft": 8,
            },
            {
              "fontSize": 12,
              "lineHeight": 16,
            },
          ],
        ]
      }
    />
    <Text
      style={
        [
          {
            "fontFamily": "Montserrat-Regular",
          },
          {
            "color": "#383838",
          },
          {
            "fontSize": 16,
          },
          [
            {
              "marginLeft": 4,
            },
            {
              "fontSize": 12,
              "lineHeight": 16,
            },
            {
              "color": "#FF4949",
            },
          ],
        ]
      }
    />
  </View>
  <View>
    <TextInput
      autoComplete="off"
      keyboardType="default"
      onChangeText={null}
      passwordGuidelines={false}
      placeholder=""
      secureTextEntry={false}
      style={
        [
          {
            "paddingBottom": 4,
            "paddingTop": 4,
          },
          {
            "paddingLeft": 10,
          },
          {
            "backgroundColor": "#FFFFFF",
          },
          {
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
          },
        ]
      }
    />
  </View>
</View>
`;
